nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureDataLakeStorage.java

```diff
@@ -212,25 +212,38 @@ static void uploadContent(DataLakeFileClient fileClient, InputStream in, long le
         fileClient.flush(length, true);
     }
 
-    //Visible for testing
+    /**
+     * This method serves as a "commit" for the upload process. To support various Conflict Resolution Strategies the processor uploads
+     * the content of the FlowFile to a temporary file with a unique name, then attempts to rename it. It is not an efficient approach,
+     * especially for large files, but it is needed because of the issue (azure-sdk-for-java/issues/31248) linked above.
```

Comment:
The temporary file + rename was needed because the "put" in ADLS is not atomic. You create the file first (0-byte file), then append the payload. So the work-in-progress file would be available for readers before the full upload is finished.
So it is not strictly related to conflict resolution and has nothing to do with issue 31248 (which is about chunked uploading of large files).

Could you please correct the documentation accordingly?

Reply:
Hey @turcsanyip!
Thanks for the review. It seems like I misinterpreted one of the comments in the linked issue. Fixed the doc!